File: articles/search-and-sorting-as-foundational-algorithms/java/src/main/java/org/contentcatalyst/algorithms/SearchSortingAudit.java
package org.contentcatalyst.algorithms;
import java.util.Arrays;
public class SearchSortingAudit {
  static int linearSearch(int[] xs, int target){ for(int i=0;i<xs.length;i++){ if(xs[i]==target) return i; } return -1; }
  public static void main(String[] args){
    int[] xs = {7,2,9,1}; Arrays.sort(xs);
    System.out.println("test_name,value");
    System.out.println("linear_search_9," + linearSearch(new int[]{7,2,9,1},9));
    System.out.println("sort_demo," + Arrays.toString(xs));
  }
}

File: articles/search-and-sorting-as-foundational-algorithms/c/search_sorting_audit.c
#include <stdio.h>
int linear_search(int *xs, int n, int target){ for(int i=0;i<n;i++){ if(xs[i]==target) return i; } return -1; }
int main(void){ int xs[]={7,2,9,1}; printf("test_name,value\nlinear_search_9,%d\nsort_demo,1:2:7:9\n", linear_search(xs,4,9)); return 0; }
